core/src/main/java/org/apache/spark/util/collection/unsafe/sort/UnsafeSorterSpillReader.java

```diff
@@ -73,7 +76,9 @@ public void loadNext() throws IOException {
     numRecordsRemaining--;
     if (numRecordsRemaining == 0) {
       in.close();
-      file.delete();
+      if (!file.delete()) {
+        logger.error("Unable to delete spill file {}", file.getPath());
```

Comment:
That grep assumes the `File` reference name ends in "file". I see 20-30 non-test usages of the method. It may not be worth logging a warning for every possible failure, but worth surveying the usages to see if any really ought to log something in case it doesn't delete.